EventsExpress/ClientApp/src/components/profile/user-profile.js

```diff
@@ -171,74 +142,81 @@ export default class UserItemView extends Component {
                             label="Visited events"
                             icon={
                                 <IconButton
                                     color={this.state.value === 2 ? '' : 'primary'}>
                                     <i className="fas fa-history"></i>
                                 </IconButton>}
-                            {...a11yProps(2)}
-                        />
+                            component={Link}
+                            to={`/user/${userId}/VisitedEvents`} />
                         <Tab
                             label="Events to go"
                             icon={
                                 <IconButton
                                     color={this.state.value === 3 ? '' : 'primary'}>
                                     <i className="fas fa-map-marker-alt"></i>
                                 </IconButton>}
-                            {...a11yProps(3)}
-                        />
-                        
-                        
+                            component={Link}
+                            to={`/user/${userId}/EventsToGo`} />
                     </Tabs>
                 </AppBar>
-                <TabPanel value={this.state.value} index={0}>
-                </TabPanel>
-                <TabPanel value={this.state.value} index={1}>
-                </TabPanel>
-                <TabPanel value={this.state.value} index={2}>
-                </TabPanel>
-                <TabPanel value={this.state.value} index={3}>
-                </TabPanel>
-                {this.props.add_event_flag ?
-                    <div className="shadow mb-5 bg-white rounded">
-                        <AddEventWrapper onCreateCanceling={() => this.handleChange(null, 0)} />
-                    </div>
-                    :
-                    <div className="shadow pl-2 pr-2 pb-2 mb-5 bg-white rounded">
-                        {spinner}{content}
-                        {!isPending && !(data.items && data.items.length > 0) &&
-                            <div id="notfound" className="w-100">
-                                <div className="notfound">
-                                    <div className="notfound-404">
-                                        <div className="h1">No Results</div>
-                                    </div>
-                                </div>
-                            </div>}
-                    </div>
-                }
+                <Switch>
+                    <Route
+                        exact
+                        path='/user/:id'
+                        render={() =>
+                            <Redirect to={`/user/${userId}/FutureEvents`} />} />
+                    <Route
+                        path='/user/:id/FutureEvents'
+                        render={() => <Events
+                            events={this.props.events}
+                            current_user={this.props.current_user}
```

Comment:
Is `current_user` prop used anywhere in the child?

Reply:
Yes, in child file it passes to EventsForProfile props to render user's events.